zone/client_process.cpp

```diff
@@ -1691,11 +1696,12 @@ void Client::OPGMTrainSkill(const EQApplicationPacket *app)
 
 		uint16 skilllevel = GetRawSkill(skill);
 
-		if(skilllevel == 0) {
+		if(skilllevel == 0 || (skilllevel == 254 && SkillTrainLevel(skill, GetClass()) <= GetLevel())) {
```

Comment:
What does checking for `254` do here exactly?

Reply:
Thank you, this was a carry over from the older TAKP code our server used. I do not believe this portion is needed anymore.

Reply:
Still need to remove this if we don't need this.